src/packages/version-check/src/version-check.ts

```diff
@@ -0,0 +1,323 @@
+   * to false.
+   */
+  disable() {
+    this.destroy();
+    // squelch errors - writes to fs.
+    try {
+      this._updateConfig({ enabled: false });
+    } catch {}
+  }
+
+  /**
+   * Closes and destroys any active requests.
+   */
+  destroy() {
+    if (this._request) {
+      this._request.close();
+    }
+
+    if (this._session) {
+      this._session.close();
+    }
+
+    this._request = null;
+    this._session = null;
+  }
+
+  /**
+   * Checks whether the required parameters pass for
+   * notifying the user of a newer version.
+   *
+   * @returns boolean
+   */
+  canNotifyUser(): boolean {
+    return (
+      !!this._currentVersion &&
+      this.isEnabled &&
+      !this.alreadyLoggedLatestVersion() &&
+      this.notificationIntervalHasPassed()
+    );
+  }
+
+  /**
+   * Returns true if the user has already been notified of the
+   * `latestVersion` or higher.
+   *
+   * @returns boolean
+   */
+  alreadyLoggedLatestVersion(): boolean {
+    return semverGte(
+      this._config.lastVersionLogged,
+      this._config.latestVersion
+    );
+  }
+
+  /**
+   * Returns true if the hardcoded `_notificationInterval` amount of
+   * time has passed since the last time we notified the user.
+   *
+   * @returns boolean
+   */
+  notificationIntervalHasPassed(): boolean {
+    const timePassed = Date.now() - this._config.lastNotification;
+    return timePassed > this._notificationInterval;
+  }
+
+  /**
+   * Makes the request and updates the config with the latest semver.
+   */
+  async getLatestVersion() {
+    if (this.isEnabled) {
+      // squelch errors - Conf writes to the fs, _fetchLatest can reject.
+      try {
+        this._updateConfig({
+          latestVersion: await this._fetchLatest()
+        });
+      } catch {}
+    }
+  }
+
+  /**
+   * Makes the request to config.url for the latest semver string for
+   * `packageName`.
+   *
+   * @returns Promise
+   */
+  private _fetchLatest(): Promise<string> {
+    this.destroy();
+    const { packageName, url, ttl } = this._config;
+
+    return new Promise<string>((resolve, reject) => {
+      const session = http2.connect(url);
+      const req = session.request({
+        ":path": `/version?package=${packageName}`
+      });
+
+      this._session = session;
+      this._request = req;
+
+      session.on("error", err => reject(err));
+      req.setEncoding("utf8");
+
+      req
+        .on("error", reject)
+        .on("response", (_headers, _flags) => {
+          let data: string = "";
+          req
+            .on("data", chunk => {
+              data += chunk;
+            })
+            .on("end", async () => {
+              resolve(data);
+              session.close();
+            });
+        })
+        .setTimeout(ttl, () => {
+          this.destroy();
+          reject(`ttl expired: ${ttl}`);
+        });
+      req.end();
+    });
+  }
+
+  /**
+   * Shows the banner message to the user on ganache startup.
+   *
+   */
+  log() {
```

Comment:
Yeah they're meant to be different. `cliMessage` is meant to always return the one-liner and is intended for features like `'-h` or `-v`. `--version`. 

It could be compelling to just return the `semverUpgradeType` and let the consumer determine what the do with it. Or maybe something structured like an object with meta data. 

The `log` function is meant to be our conditional banner message. 

`cliMessage` -> Many times, not bound to any conditions, one-liner message, called externally.
`log` -> Once per new version, per 24 hour window, banner message, called externally & controlled internally.

I think `cliMessage` is a terrible name. Something like `upgradeType` or `upgradeMessage` may be prudent with `upgradeBannerMessage`